dkc/core/rest/file.py

```diff
@@ -83,11 +101,42 @@ def get_serializer_class(self):
             return FileUpdateSerializer
         return FileSerializer
 
+    def _validate_authorized_upload(self, authorization: str, folder: Folder) -> User:
+        try:
+            signed_obj = signing.loads(authorization)
+        except signing.BadSignature:
+            logger = logging.getLogger('django.security.SuspiciousOperation')
+            logger.warning('Authorized upload signature tampering detected.')
+            raise PermissionDenied('Invalid authorization signature.')
+
+        if signed_obj.get('scope') != 'authorized_upload':
+            raise PermissionDenied('Invalid signature scope.')
```

Comment:
This is probably also a 400. It's invalid input; they passed the wrong (validly signed) type of token.